pyproject.toml

```diff
@@ -0,0 +1,51 @@
+[tool.poetry]
+name = "perfect-todo-app"
+version = "0.1.0"
+description = ""
+authors = ["Perry Jones <perry.jones@softwire.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.8"
+astroid = "2.4.1"
+atomicwrites = "1.4.0"
+attrs = "19.3.0"
+certifi = "2020.4.5.2"
+chardet = "3.0.4"
+click = "7.1.2"
+colorama = "0.4.3"
+dominate = "2.5.1"
+Flask = "1.1.1"
+Flask-Bootstrap = "3.3.7.1"
+Flask-Datepicker = "0.13"
+gunicorn = "20.0.4"
+idna = "2.9"
+isort = "4.3.21"
+itsdangerous = "1.1.0"
+Jinja2 = "2.11.2"
+lazy-object-proxy = "1.4.3"
+MarkupSafe = "1.1.1"
+mccabe = "0.6.1"
+more-itertools = "8.4.0"
+packaging = "20.4"
+pluggy = "0.13.1"
+py = "1.9.0"
+pylint = "2.5.2"
+pyparsing = "2.4.7"
+pytest = "5.4.3"
+python-dotenv = "0.12.0"
+requests = "2.24.0"
+rope = "0.17.0"
+selenium = "3.141.0"
```

Comment:
You should move pytest, selenium etc into dev-dependencies section.

Also, you do not need to include nested dependencies. I expect you to be able to trim this list down to around 5 dependencies, plus two or three test ones.

Reply:
I've also optimised this on exercise-5 branch too